### PR TITLE
DEVPROD-11682 Remove GlobalGitHubTokenExpansion

### PR DESCRIPTION
**Pull request by malikchaya2**
DEVPROD-11682 

### Description
This is no longer needed 

### Testing
existing tests

### CODE REVIEW

File: agent/command/git_test.go [hunk 1]
```diff
@@ -452,7 +422,8 @@ func (s *GitGetProjectSuite) TestValidateGitCommands() {
 	defer cancel()
 	var pluginCmds []Command
 
-	s.comm.CreateGitHubDynamicAccessTokenResult = mockedGitHubAppToken
+	// s.comm.CreateGitHubDynamicAccessTokenResult = mockedGitHubAppToken
+	// s.comm.CreateInstallationTokenResult = mockedGitHubAppToken
```
Comment (Kimchelly) [inline agent/command/git_test.go:426]: Can this be removed? Or is it still needed?

File: agent/internal/client/base_client.go [hunk 2]
```diff
@@ -964,6 +964,10 @@ func (c *baseCommunicator) CreateGitHubDynamicAccessToken(ctx context.Context, t
 	return token.Token, nil
 }
 
+func (c *baseCommunicator) CreateCachedInstallationToken(context.Context, string, string, time.Duration, *github.InstallationTokenOptions) (string, error) {
```
Comment (Kimchelly) [inline agent/internal/client/base_client.go:967]: Can remove this since it's unused code.

**Review by Kimchelly — CHANGES_REQUESTED** [review]
* Can we remove [the oauth token parameter](https://github.com/evergreen-ci/evergreen/blob/159e1e949078e78201cb49b93cd6b653b4ca17bf/model/project.go#L1016) from the expansions since it should now be unused?
* Can we delete the `GlobalGitHubTokenExpansion` constant since it should be unused?
* Should [this doc](https://github.com/evergreen-ci/evergreen/blob/b5fa0e3a78228f51f5fbcbf354cdd920d234bc45/docs/Project-Configuration/Project-Commands.md?plain=1#L456) be updated since the expansion will no longer exist?

**Comment by malikchaya2** [comment]
> * Can we remove [the oauth token parameter](https://github.com/evergreen-ci/evergreen/blob/159e1e949078e78201cb49b93cd6b653b4ca17bf/model/project.go#L1016) from the expansions since it should now be unused?
> * Can we delete the `GlobalGitHubTokenExpansion` constant since it should be unused?
> * Should [this doc](https://github.com/evergreen-ci/evergreen/blob/b5fa0e3a78228f51f5fbcbf354cdd920d234bc45/docs/Project-Configuration/Project-Commands.md?plain=1#L456) be updated since the expansion will no longer exist?

Yes, I picked these changes out of a larger change and left a few behind accidentally. Thank you!

File: agent/command/git.go [hunk 3]
```diff
@@ -160,40 +160,16 @@ func getProjectMethodAndToken(ctx context.Context, comm client.Communicator, td
 	owner := conf.ProjectRef.Owner
 	repo := conf.ProjectRef.Repo
 	appToken, err := comm.CreateInstallationToken(ctx, td, owner, repo)
+	if err != nil {
+		return "", "", errors.Wrap(err, "creating app token")
+	}
 	if appToken != "" {
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply agent/command/git.go:166]: I think we can assume that no errors means a token was generated. On top of that, there may be situations where things will still work without one like cloning public repos.

**Review by Kimchelly — APPROVED** [review]
LGTM mod remove [the now-unused `oauthToken` parameter for `PopulateExpansions`](https://github.com/evergreen-ci/evergreen/blob/75b190c63859657f5f808219e5432d891c33c973/model/project.go#L1004).

**Pull request merged**